homeassistant/components/deluge/sensor.py

```diff
@@ -1,128 +1,100 @@
+
+from .const import DOMAIN, SENSOR_TYPES
 
 _LOGGER = logging.getLogger(__name__)
-_THROTTLED_REFRESH = None
-
-DEFAULT_NAME = "Deluge"
-DEFAULT_PORT = 58846
-DHT_UPLOAD = 1000
-DHT_DOWNLOAD = 1000
-SENSOR_TYPES = {
-    "current_status": ["Status", None],
-    "download_speed": ["Down Speed", "kB/s"],
-    "upload_speed": ["Up Speed", "kB/s"],
-}
-
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_HOST): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_MONITORED_VARIABLES, default=[]): vol.All(
-            cv.ensure_list, [vol.In(SENSOR_TYPES)]
-        ),
-    }
-)
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Set up the Deluge sensors."""
-    from deluge_client import DelugeRPCClient
-
-    name = config.get(CONF_NAME)
-    host = config.get(CONF_HOST)
-    username = config.get(CONF_USERNAME)
-    password = config.get(CONF_PASSWORD)
-    port = config.get(CONF_PORT)
-
-    deluge_api = DelugeRPCClient(host, port, username, password)
-    try:
-        deluge_api.connect()
-    except ConnectionRefusedError:
-        _LOGGER.error("Connection to Deluge Daemon failed")
-        raise PlatformNotReady
+
+
+async def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Import config from configuration.yaml."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up Deluge sensors."""
+
+    client = hass.data[DOMAIN][config_entry.entry_id]
+    name = config_entry.data[CONF_NAME]
+
     dev = []
-    for variable in config[CONF_MONITORED_VARIABLES]:
-        dev.append(DelugeSensor(variable, deluge_api, name))
+    for sensor_type in SENSOR_TYPES:
+        dev.append(DelugeSensor(sensor_type, client, name,))
 
-    add_entities(dev)
+    async_add_entities(dev, True)
 
 
 class DelugeSensor(Entity):
     """Representation of a Deluge sensor."""
 
-    def __init__(self, sensor_type, deluge_client, client_name):
+    def __init__(self, sensor_type, client, client_name):
         """Initialize the sensor."""
         self._name = SENSOR_TYPES[sensor_type][0]
-        self.client = deluge_client
+        self.client = client
         self.type = sensor_type
         self.client_name = client_name
         self._state = None
         self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
         self.data = None
-        self._available = False
+        self.unsub_dispatcher = None
 
     @property
     def name(self):
         """Return the name of the sensor."""
         return f"{self.client_name} {self._name}"
 
+    @property
+    def unique_id(self):
+        """Return the unique id of the entity."""
+        return f"{self.client.api.host}-{self.name}"
```

Comment:
Names should not be part of unique ID. Can you fetch some instance identifier for Deluge ?

Reply:
The identifier for Deluge is the host name. `self.name` is to identify the entity (sensor or switch). As this integration doesn't create devices this is what I though off to link the entities to the integration.

Reply:
Can we get a unique identifier for the deluge instance that is not the host name? 

The problem with name is that it can change, making it not very unique. Same with hostname.

Reply:
I understand but for this integration deluge will not provide a unique identifier. I will use self.type (as this is fixed) and in case the host name changes (which will require deleting and recreating the entry) I guess the user will not get any history for those entities, which should be ok for this type of integration.